### ISSUE TITLE
[pallet-referenda] Crowd Sourced Decision Deposit

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#6041]
Currently the decision deposit for a referendum must be paid in full by a single address. We should add a mechanism that allows it to be crowdsourced. @bkchr suggested that we can re-use the code of the [crowdloan-pallet](https://github.com/paritytech/polkadot-sdk/blob/7ecf3f757a5d6f622309cea7f788e8a547a5dce8/polkadot/runtime/common/src/crowdloan/mod.rs#L174) from the Polkadot runtime. I am not familiar with the code of tha pallet but probably needs a new pallet though to manage the deposits.

Requirement:
- It should be possible to split the payment between different addresses (at least 128)
- The code should only consider the `n` highest amounts paid and refund the lower ones

**Comment by runcomet** [comment]
Hi @ggwpez, would like to work on this! Decision deposits might mitigate some whale activity in OpenGov

Assigning me would be nice if possible

**Comment by ggwpez** [comment]
Thanks @runcomet!

**Comment by bkchr** [comment]
I already have this ready in a branch 😅

**Comment by bkchr** [comment]
https://github.com/paritytech/polkadot-sdk/tree/bkchr-referenda-deposit-contribute this is my current work. Just misses some benchmarking improvements and more tests.

**Comment by apopiak** [comment]
> https://github.com/paritytech/polkadot-sdk/tree/bkchr-referenda-deposit-contribute this is my current work. Just misses some benchmarking improvements and more tests.

Do you want help cleaning it up? (Looking for bounties projects for PBA graduates)

**Comment by bkchr** [comment]
Yeah can be taken by someone @apopiak. If you have someone, you can send them to me.

**Comment by PolkadotDom** [comment]
@apopiak Will take this if you haven't passed it off yet

**Comment by apopiak** [comment]
Go for it

**Comment by runcomet** [comment]
@PolkadotDom Still working on this?
